gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AllergyIntoleranceMapper.java

```diff
@@ -0,0 +1,167 @@
+        var observationStatement =
+            compoundStatement.getComponent().stream().map(RCMRMT030101UK04Component02::getObservationStatement).findFirst().get();
+        var codeableConceptFromCode = codeableConceptMapper.mapToCodeableConcept(observationStatement.getCode());
+        var compoundCode = compoundStatement.getCode().getCode();
+
+        if (NON_DRUG_ALLERGY_CODE.equals(compoundCode)) {
+            allergyIntolerance.setCode(codeableConceptFromCode);
+        }
+
+        if (DRUG_ALLERGY_CODE.equals(compoundCode)) {
+            if (observationStatement.hasValue() && observationStatement.getValue() instanceof CD value) {
+                if (!CODE_SYSTEM.equals(value.getCodeSystem())) {
+                    var codeableConceptFromValue = codeableConceptMapper.mapToCodeableConcept(value);
+                    allergyIntolerance.setCode(codeableConceptFromValue);
+
+                    var codeDisplayName = codeableConceptFromCode.getCodingFirstRep().getDisplay();
+                    if (codeDisplayName != null && !ALLERGY_TERM_TEXT.equals(codeDisplayName)
+                        && !codeDisplayName.equals(codeableConceptFromValue.getCodingFirstRep().getDisplay())) {
+                        allergyIntolerance.getNote().add(new Annotation().setText(ALLERGY_NOTE.formatted(codeDisplayName)));
+                    }
+                } else {
+                    allergyIntolerance.setCode(codeableConceptFromCode);
+                }
+            } else {
+                allergyIntolerance.setCode(codeableConceptFromCode);
+            }
+        }
+    }
+
+    private void buildOnset(RCMRMT030101UK04CompoundStatement compoundStatement, AllergyIntolerance allergyIntolerance) {
+        var effectiveTime = compoundStatement.getEffectiveTime();
+        var availabilityTime = compoundStatement.getAvailabilityTime();
+
+        if (effectiveTime.hasLow() && effectiveTime.getLow().hasValue()) {
+            allergyIntolerance.setOnset(parseToDateTimeType(effectiveTime.getLow().getValue()));
+        } else if (effectiveTime.hasCenter() && effectiveTime.getCenter().hasValue()) {
+            allergyIntolerance.setOnset(parseToDateTimeType(effectiveTime.getCenter().getValue()));
+        } else if (availabilityTime.hasValue()) {
+            allergyIntolerance.setOnset(parseToDateTimeType(availabilityTime.getValue()));
+        }
+    }
+
+    private DateTimeType getAssertedDateElement(RCMRMT030101UK04EhrExtract ehrExtract, RCMRMT030101UK04EhrComposition ehrComposition) {
+        return ehrComposition.hasAuthor() && ehrComposition.getAuthor().hasTime()
+            ? parseToDateTimeType(ehrComposition.getAuthor().getTime().getValue())
+            : parseToDateTimeType(ehrExtract.getAvailabilityTime().getValue());
+    }
+
+    private void buildNote(AllergyIntolerance allergyIntolerance, RCMRMT030101UK04CompoundStatement compoundStatement) {
+        compoundStatement.getComponent().get(0)
```

Comment:
Will the observation statement always be at index 0? is it possible for it to be in another position?